Source/WebKit/NetworkProcess/ServiceWorker/WebSWServerToContextConnection.cpp

```diff
@@ -131,8 +132,20 @@ void WebSWServerToContextConnection::firePushEvent(ServiceWorkerIdentifier servi
     if (data)
         ipcData = IPC::DataReference { data->data(), data->size() };
     sendWithAsyncReply(Messages::WebSWContextManagerConnection::FirePushEvent(serviceWorkerIdentifier, ipcData), [weakThis = WeakPtr { *this }, callback = WTFMove(callback)](bool wasProcessed) mutable {
```

Comment:
Another issue is that we should be calling `setMostRecentWebPushInteractionTime` when the `notificationclick` event is handled successfully in the service worker, and not when the `push` event occurs.